planner/core/expression_rewriter.go

```diff
@@ -1175,6 +1177,34 @@ func (er *expressionRewriter) betweenToExpression(v *ast.BetweenExpr) {
 // Otherwise it should return false to indicate (the caller) that original behavior needs to be performed.
 func (er *expressionRewriter) rewriteFuncCall(v *ast.FuncCallExpr) bool {
 	switch v.FnName.L {
+	// when column is not null, ifnull on such column is not necessary.
+	case ast.Ifnull:
+		if er.isJoin {
```

Comment:
what does this mean? we don't eliminate the `ifnull` function inside a `join` operator?